int64 point1_x
int64 point1_y
int64 point2_x
int64 point2_y
int64 point3_x
int64 point3_y
int64 point4_x
int64 point4_y
